src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs

```diff
@@ -779,29 +779,23 @@ private static string RetrieveProcessUserName(Process process)
 
                     var tokenUser = Marshal.PtrToStructure<Win32Native.TOKEN_USER>(tokenUserInfo);
 
-                    // Set the default length to be 256, so it will be sufficient for most cases.
-                    int userNameLength = 256, domainNameLength = 256;
-                    var userNameStr = new StringBuilder(userNameLength);
-                    var domainNameStr = new StringBuilder(domainNameLength);
+                    // Max username is defined as UNLEN = 256 in lmcons.h
+                    // Max domainname is defined as DNLEN = CNLEN = 15 in lmcons.h
+                    // The buffer length must be +1, last position is for a null string terminator.
+                    int userNameLength = 257;
+                    int domainNameLength = 16;
+                    Span<char> userNameStr = stackalloc char[userNameLength];
+                    Span<char> domainNameStr = stackalloc char[domainNameLength];
                     Win32Native.SID_NAME_USE accountType;
 
                     if (!Win32Native.LookupAccountSid(null, tokenUser.User.Sid, userNameStr, ref userNameLength, domainNameStr, ref domainNameLength, out accountType))
```

Comment:
Could you please leave a comment to call out that `userNameLength` and `domainNameLength` will be set to the actual length of the user name and domain name respectively?

Reply:
Done.